tasks/basecalling/task_dorado_demux.wdl

```diff
@@ -0,0 +1,102 @@
+version 1.0
+
+task dorado_demux {
+  input {
+    Array[File] bam_files
+    String kit_name
+    String fastq_file_name
+    Int cpu = 4 
+    Int memory = 16
+    Int disk_size = 100
+    String docker = "us-docker.pkg.dev/general-theiagen/staphb/dorado:0.8.0"
+  }
+
+  command <<< 
+    set -euo pipefail
+
+    fastq_file_name="~{fastq_file_name}"
+    kit_name="~{kit_name}"
+
+    echo "### Starting Dorado demux ###"
+    date
+    echo "Input BAM files:"
+    for bam_file in ~{sep=" " bam_files}; do echo "$bam_file"; done
+
+    # Process each BAM file in a unique output directory
+    for bam_file in ~{sep=" " bam_files}; do
+      base_name=$(basename "$bam_file" .bam)
+      demux_dir="demux_output_${base_name}"
+      mkdir -p "$demux_dir"
+
+      echo "Processing BAM file: $bam_file into directory $demux_dir"
+      echo "Running dorado demux for kit: $kit_name"
+
+      # Run Dorado demux command
+      dorado demux \
+        "$bam_file" \
+        --output-dir "$demux_dir" \
+        --kit-name "$kit_name" \
+        --emit-fastq \
+        --emit-summary \
+        --verbose > "$demux_dir/demux_${base_name}.log" 2>&1 || {
```

Comment:
just want to note that `dorado demux` does allow you to control threads/cpus used and that by default it will use all available threads. I don't forsee any issue with this, unless running on a shared compute system (i.e. not Terra). I don't think it's necessary to update the code for this now

```
-t, --threads          Combined number of threads for barcoding and output generation. Default uses all available threads. [nargs=0..1] [default: 0]
```